src/CoreBundle/private/modules/CMSTemplateEngine/CMSTemplateEngine.class.php

```diff
@@ -108,18 +111,29 @@ public function Init()
         parent::Init();
 
         $this->sPageId = $this->getPageIdFromRequest();
+
+        // TODO what about language?
+        $page = $this->getPageService()->getById($this->sPageId);
+
+        if (null === $page) {
+            throw new NotFoundHttpException(sprintf('A page with the id %s cannot be found.', $this->sPageId));
```

Comment:
As there is (probably) no notion of "404" in the backend this is also problematic.

Reply:
404 is o.k. here, but we need a new issue about adding 404 handling to the backend controller.

Reply:
Entere #527 for 404.